### DISCUSSION TITLE
atplot of a single cell, without cavity

### DISCUSSION THREAD

**Discussion by simoneliuzzo** [atcollab/at#440] [General]
The command atplot is returning an error (missing cavity) when producing a plot of a single cell extracted from a lattice with radiation on. The error is correct as the lattice is not complete and in fact it does not only miss a cavity, but all the other cells to close the lattice. 

Nevertheless, for a single cell, 4D may be assumed implicitly?
 
In general, when working with an arbitrary lattice (4d or 6d), I must replace the usual friendly call to atplot(ARC) with atplot(atradoff(ARC)) to make sure the plot will appear.

**Comment by lfarv** [comment]
Hi @simoneliuzzo.
`atplot` is just displaying the result of `atlinopt6`. So you question should rather be addressed to `atlinopt6`.

`atlinopt6` (in "closed ring" mode, not in "transfer line mode") only needs a closed solution (exactly as `atlinopt`). It has no notion of arc, cell, ring… and it does not care. It works perfectly with a lattice without any bending magnet, which cannot be neither an arc nor a full ring. The only condition to get &beta; values is to find a closed solution, either in 4D or in 6D.

So it's perfectly valid to work in 6D on a single superperiod. It's even the most efficient solution for studying periodic rings: same result with less computation. 

Your problem probably comes when giving a cell with radiation but no cavity. Then there is obviously no closed solution ! Either you add a cavity, or you remove the radiation, but it's up to you to decide, AT cannot guess what you want.

Notes:
- even trying to use the old `atlinopt` with radiation (which is strictly forbidden) will never give a solution in these conditions. So no regret, it never worked !
- using `atplot(atradoff(ring))` is not a bad solution: &beta;-functions don't depend on longitudinal motion. `atplot` now accepts 6D lattices for your convenience: it works with any arbitrary **stable** lattice, 4D or 6D. But 6D has strictly no added value for the plot.
